tests/v2/validation/upgrade/kubernetes.go

```diff
@@ -73,6 +73,15 @@ func upgradeLocalCluster(u *suite.Suite, testName string, client *rancher.Client
 		err = waitForLocalClusterUpgrade(client, clusterMeta.ID)
 		require.NoError(u.T(), err)
 
+		err = kwait.PollUntilContextTimeout(context.TODO(), 2*time.Second, defaults.FiveSecondTimeout, true, func(ctx context.Context) (done bool, err error) {
+			isConnected, err := client.IsConnected()
+			if err != nil {
+				return false, err
+			}
+			return isConnected, nil
+		})
+		require.NoError(u.T(), err)
```

Comment:
why not add this to the waitForLocalClusterUpgrade func instead of having it here?

Reply:
There was a downtime where the client was unreachable after the local cluster upgrades, therefore I added it here instead of the waitForLocalUpgrade.. but sure, we can move it to wait for local cluster upgrade.